File: src/main/java/com/fiserv/luc/api/domain/service/brazil/BRPhysicPersonService.java
package com.fiserv.luc.api.domain.service.brazil;

import com.fiserv.luc.api.domain.entity.brazil.BRPhysicPerson;
import com.fiserv.luc.api.domain.repository.brazil.BRPhysicPersonRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.history.Revision;
import org.springframework.data.history.Revisions;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class BRPhysicPersonService {

    /**
     *
     */
    private final BRPhysicPersonRepository physicPersonRepository;

    /**
     * @param filters  String
     * @param pageable Pageable
     * @return Page<BRPhysicPerson>
     */
    public Page<BRPhysicPerson> listByFilters(final String filters, final Pageable pageable) {
        return this.physicPersonRepository.findAll(pageable);
    }

    /**
     * @param id Long
     * @return Optional<BRPhysicPerson>
     */
    public Optional<BRPhysicPerson> findById(final Long id) {
        return this.physicPersonRepository.findById(id);
    }

    /**
     * @param id Long
     * @return Revisions<Long, BRPhysicPerson>
     */
    public List<Revision<Long, BRPhysicPerson>> findRevisionsById(final Long id) {
        return this.physicPersonRepository.findRevisions(id).getContent();
    }
}

File: src/main/java/com/fiserv/luc/api/domain/service/brazil/BRPhysicPersonService.kt
package com.fiserv.luc.api.domain.service.brazil

import com.fiserv.luc.api.domain.entity.brazil.BRPhysicPerson
import com.fiserv.luc.api.domain.repository.brazil.BRPhysicPersonRepository
import com.fiserv.luc.api.infrastructure.database.audited.revision.AbstractRevisionService
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service
import java.util.*

@Service
class BRPhysicPersonService(private val physicPersonRepository: BRPhysicPersonRepository) : AbstractRevisionService<BRPhysicPerson?>() {

    /**
     * @param filters  String
     * @param pageable Pageable
     * @return Page<BRPhysicPerson>
    </BRPhysicPerson> */
    fun listByFilters(filters: String?, pageable: Pageable?): Page<BRPhysicPerson> {
        return physicPersonRepository.findAll(pageable)
    }

    /**
     * @param id Long
     * @return Optional<BRPhysicPerson>
    </BRPhysicPerson> */
    fun findById(id: Long): Optional<BRPhysicPerson> {
        return physicPersonRepository.findById(id)
    }

    /**
     * @param physicPerson BRPhysicPerson
     * @return BRPhysicPerson
     */
    fun save(physicPerson: BRPhysicPerson): BRPhysicPerson {
        return physicPersonRepository.save(physicPerson)
    }

    /**
     * @param id Long
     */
    fun deleteById(id: Long) {
        physicPersonRepository.deleteById(id)
    }
}
